rt_diversity_filter.go

```diff
@@ -0,0 +1,101 @@
+package dht
+
+import (
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/libp2p/go-libp2p-kbucket/peerdiversity"
+	ma "github.com/multiformats/go-multiaddr"
+	"sync"
+)
+
+var _ peerdiversity.PeerIPGroupFilter = (*rtPeerIPGroupFilter)(nil)
+
+// locking is the responsibility of the caller
```

Comment:
what are you trying to convey here? There is a mutex in the struct that's taken by the relevant exported functions.

Reply:
Remanant. Have removed this.